packages/trader/src/Modules/Trading/Containers/__tests__/trade-params-mobile.spec.tsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mockStore } from '@deriv/stores';
+import { TCoreStores } from '@deriv/stores/types';
+import TradeParamsModal, { LastDigitMobile, BarrierMobile } from '../trade-params-mobile';
+import TraderProviders from '../../../../trader-providers';
+
+const mock_default_props = {
+    is_open: true,
+    toggleModal: jest.fn(),
+    tab_index: 0,
+};
+
+const default_mock_store = {
+    modules: {
+        trade: {
+            amount: 10,
+            basis_list: [{ text: 'basis', value: '1' }],
+            basis: '1',
+            duration: 12301911,
+            duration_unit: 'm',
+            duration_units_list: [{ text: '4', value: 'm' }],
+            expiry_epoch: '123762300',
+            form_components: ['duration', 'amount'],
+            is_vanilla: false,
+            is_turbos: false,
+        },
+    },
+};
+
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Modal: jest.fn(({ children, height }) => <div data-testid={height}>{children}</div>),
+    ThemedScrollbars: jest.fn(({ children }) => <div>{children}</div>),
+    Div100vhContainer: jest.fn(({ children }) => <div>{children}</div>),
+    Tabs: jest.fn(({ children }) => <div>{children}</div>),
+}));
+jest.mock('Modules/Trading/Components/Form/TradeParams/amount-mobile', () =>
+    jest.fn(props => (
```

Comment:
Looks like you are already mocking the whole component. Might as well not mock and use the actual component

Reply:
Hi, I didn't want to use real component because:
 1) it renders inside <Basis /> and <Basis /> renders <MinMaxStakeInfo /> and <Numpad /> and etc. Some of them are connected to the store, which will make test a bit messy if I add to the default_mock_store.modules.trade all variables used in children.
 2) this is unit test, I'm trying to avoid integration tests in such cases.
 3) such test is easy to maintain as you are independent from children.
 4) no need to refactor after each change of one of the child components
 5) even if the tests will fail, you'll always know in which component the problem is (no need in research between children components)